src/components/inputs/checkboxes/CheckboxTree/CheckboxTree.tsx

```diff
@@ -575,21 +582,35 @@ function createIsLeafVisible<T>(
   getNodeId: CheckboxTreeProps<T>['getNodeId'],
   getNodeChildren: CheckboxTreeProps<T>['getNodeChildren'],
   isAdditionalFilterApplied: CheckboxTreeProps<T>['isAdditionalFilterApplied'],
-  isSearchable: CheckboxTreeProps<T>['isSearchable']
+  isSearchable: CheckboxTreeProps<T>['isSearchable'],
+  filteredList: CheckboxTreeProps<T>['filteredList'],
 ) {
-  // if not searching, then all nodes are visible
-  if (!isActiveSearch(isAdditionalFilterApplied, isSearchable, searchTerm)) {
+  // if not searching, no additional filters are applied, and filteredList is undefined, then all nodes are visible
+  if (!isActiveSearch(isAdditionalFilterApplied, isSearchable, searchTerm) && !filteredList) {
     return (nodeId: string) => true;
   }
   // otherwise must construct array of visible leaves
   const visibleLeaves = new Set<string>();
   const searchTerms = parseSearchQueryString(searchTerm);
+  const filteredSet = new Set(filteredList);
   const addVisibleLeaves = (node: T, parentMatches: boolean) => {
-    // if parent matches, automatically match (always show children of matching parents)
-    const nodeMatches = (parentMatches || searchPredicate(node, searchTerms));
+    const nodeId = getNodeId(node);
+    let nodeMatches = false;
+    if (parentMatches) {
+      // if parent matches, automatically match (always show children of matching parents)
+      nodeMatches = parentMatches;
+    } else if (searchTerm && filteredList && filteredList.length) {
```

Comment:
I guess I was also thinking this would change to:
```suggestion
    } else if (searchTerm && filteredList) {
```

Same with line 604.

I realize that means the user might not see anything, if `filteredList` is empty, but that might be better than seeing everything?

Reply:
But wouldn't it actually be _**guaranteed**_ that a user wouldn't see anything if `filteredList` is an empty array, irrespective of a search term? I'm struggling to understand why we'd want that behavior other than some sort of workaround to render an empty tree. Of course, in the case of the related `SiteSearch` work I could implement the behavior I want by passing in either an array with values or undefined, so there's still flexibility by the consumer. I just can't quite wrap my head around why we'd want this behavior in the first place.

Reply:
I see where you are coming from. My thinking is that `filteredList` is the result of some filtering done upstream of the component. Maybe there are multiple toggles the user can turn on, and some combination of them results in an empty array. In that case, I think we would want to respect that and display an empty tree.